frontend/frontend.go

```diff
@@ -311,13 +313,13 @@ func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request
 
 	parsed, _ := azure.ParseResourceID(resourceID)
 	var doc *HCPOpenShiftClusterDocument
-	doc, found, err := f.dbClient.GetClusterDoc(ctx, resourceID, parsed.SubscriptionID)
+	doc, err = f.dbClient.GetClusterDoc(ctx, resourceID, parsed.SubscriptionID)
 	if err != nil {
 		f.logger.Error("failed to fetch document for %s: %v", resourceID, err)
 		arm.WriteInternalServerError(writer)
 		return
 	}
-	if !found {
+	if errors.Is(err, ErrNotFound) {
```

Comment:
This check needs to come before `if err != nil` or else we'll end up treating `ErrNotFound` like any other error.

Reply:
good catch, thanks! Change coming

Reply:
ok should be updated everywhere to 
1) Check for the ErrNotFound first when it make sense to
2) Finishes implementing the change from bool to error for Subscription docs too